Bean instantiation error when Redis is disabled

### Expected Behavior

disabling redis.enabled doesn't cause exception

### Actual Behaviour

Exception occurs. Symptoms described in this PR with possible solution https://github.com/micronaut-projects/micronaut-redis/pull/629

### Steps To Reproduce

See https://github.com/micronaut-projects/micronaut-redis/pull/629

### Environment Information

_No response_

### Example Application

_No response_

### Version

4.10.x

I reproduced this on the current checkout (`678de559522af817aacbf8a755885cdd0c8b2a56`, current `master` tip in this workspace).

Environment used for validation:
- Java: GraalVM CE 25.0.2
- Gradle: 9.4.1
- OS: Linux 6.18.5 aarch64

Minimal reproducer:

```groovy
ApplicationContext applicationContext = ApplicationContext.run([
    'redis.enabled': false,
    'redis.host': 'localhost',
    'redis.port': 6379,
    'redis.caches.test.enabled': true,
    'redis.caches.test.expire-after-write': '24h'
])
applicationContext.getBean(io.micronaut.cache.DefaultCacheManager)
```

Observed result:
- `ApplicationContext.run(...)` succeeds.
- The failure happens lazily when `DefaultCacheManager` is first resolved.
- The same failure also occurs when resolving the named `RedisCache` bean directly.

Failure:

```text
io.micronaut.context.exceptions.BeanInstantiationException: Error instantiating bean of type [io.micronaut.cache.DefaultCacheManager]
Message: No Redis server configured to allow caching
Caused by: io.micronaut.context.exceptions.ConfigurationException: No Redis server configured to allow caching
    at io.micronaut.configuration.lettuce.RedisConnectionUtil.openBytesRedisConnection(RedisConnectionUtil.java:153)
    at io.micronaut.configuration.lettuce.cache.RedisCache.<init>(RedisCache.java:84)
```

Relevant code path in the current sources:
- `RedisCache` is still created for `redis.caches.*` and immediately opens a Redis connection in its constructor.
- `redis.enabled=false` prevents the Redis client/configuration beans from existing, so that constructor path fails when cache beans are resolved.

This is actionable. The issue is not limited to an old branch report; it still reproduces on the current checkout.

--- a/redis-lettuce/src/main/java/io/micronaut/configuration/lettuce/cache/RedisCache.java
+++ b/redis-lettuce/src/main/java/io/micronaut/configuration/lettuce/cache/RedisCache.java
@@ -50,6 +50,7 @@ import java.util.function.Supplier;
  * @since 1.0
  */
 @EachBean(RedisCacheConfiguration.class)
+@Requires(property = RedisSetting.PREFIX + ".enabled", notEquals = StringUtils.FALSE)
 @Requires(classes = SyncCache.class, property = RedisSetting.REDIS_POOL + ".enabled", defaultValue = StringUtils.FALSE, notEquals = StringUtils.TRUE)
 public class RedisCache extends AbstractRedisCache<StatefulConnection<byte[], byte[]>> {
     private final RedisAsyncCache asyncCache;

--- a/redis-lettuce/src/main/java/io/micronaut/configuration/lettuce/cache/RedisConnectionPoolCache.java
+++ b/redis-lettuce/src/main/java/io/micronaut/configuration/lettuce/cache/RedisConnectionPoolCache.java
@@ -53,6 +53,7 @@ import java.util.function.Supplier;
  * @since 5.3.0
  */
 @EachBean(RedisCacheConfiguration.class)
+@Requires(property = RedisSetting.PREFIX + ".enabled", notEquals = StringUtils.FALSE)
 @Requires(classes = SyncCache.class, property = RedisSetting.REDIS_POOL + ".enabled", defaultValue = StringUtils.FALSE, notEquals = StringUtils.FALSE)
 public class RedisConnectionPoolCache extends AbstractRedisCache<AsyncPool<StatefulConnection<byte[], byte[]>>> {
     private static final Logger LOG = LoggerFactory.getLogger(RedisConnectionPoolCache.class);

--- a/redis-lettuce/src/test/groovy/io/micronaut/configuration/lettuce/RedisConfigurationSpec.groovy
+++ b/redis-lettuce/src/test/groovy/io/micronaut/configuration/lettuce/RedisConfigurationSpec.groovy
@@ -1,5 +1,7 @@
 package io.micronaut.configuration.lettuce
 
+import io.micronaut.cache.DefaultCacheManager
+import io.micronaut.cache.SyncCache
 import io.lettuce.core.api.StatefulRedisConnection
 import io.lettuce.core.RedisClient
 import io.lettuce.core.RedisCredentialsProvider
@@ -51,6 +53,43 @@ class RedisConfigurationSpec extends Specification {
 
         then:
         thrown(NoSuchBeanException)
+    }
+
+    void "test DefaultCacheManager available when redis disabled and caches configured"() {
+        given:
+        applicationContext = ApplicationContext.run([
+                'redis.enabled'                     : false,
+                'redis.host'                        : 'localhost',
+                'redis.port'                        : 6379,
+                'redis.caches.test.enabled'         : true,
+                'redis.caches.test.expire-after-write': '24h'
+        ])
+
+        when:
+        DefaultCacheManager cacheManager = applicationContext.getBean(DefaultCacheManager)
+
+        then:
+        cacheManager != null
+        applicationContext.getBeansOfType(SyncCache).isEmpty()
+    }
+
+    void "test DefaultCacheManager available when redis disabled and pooled caches configured"() {
+        given:
+        applicationContext = ApplicationContext.run([
+                'redis.enabled'                     : false,
+                'redis.host'                        : 'localhost',
+                'redis.port'                        : 6379,
+                'redis.pool.enabled'                : true,
+                'redis.caches.test.enabled'         : true,
+                'redis.caches.test.expire-after-write': '24h'
+        ])
+
+        when:
+        DefaultCacheManager cacheManager = applicationContext.getBean(DefaultCacheManager)
+
+        then:
+        cacheManager != null
+        applicationContext.getBeansOfType(SyncCache).isEmpty()
     }
 
     void "test redis metrics configuration binds command latency recorder settings"() {

